apps/judicial-system/web/src/utils/hooks/use-case/useCase.tsx

```diff
@@ -313,6 +316,38 @@ const useCase = () => {
     [updateCase],
   )
 
+  const autofillSessionArrangements = useMemo(
```

Comment:
The bodies of these functions are almost identical to the `autofill` function. Would it be possible to change the value parameter in the autofill function to have type `string | SessionArrangements | boolean` and use that instead? Inside autofill we could cast the value parameter to string if needed?

Reply:
It is possible, but would be somewhat messy, as we would need to base our decisions on field names rather than field types. Also, the logic for the boolean autofill function is different from the other functions. Instead of checking for falsie current values we need to check for undefined/null.

The reason, I created a separate function for the SessionArrangements type was to get rid of a typescript error.
The reason, I created a separate function for the boolean type was the different logic.

If you don't mind, I would like to leave it like this for now and come pack to it later for possible refactoring.